R/custom_r.R

```diff
@@ -0,0 +1,25 @@
+#' @noRd
+get_custom_r <- function(path) {
+  code <- readLines(path)
+
+  if (any(grepl(pattern = "^#' @param", x = code))) {
+    cli::cli_abort(
+      c(
+        "{.code @param} tags not allowed for custom components",
+        i = "use mustache component template instead"
+      )
+    )
+  } else if (any(grepl(pattern = "\\{\\{|\\}\\}", x = code))) {
+    cli::cli_abort(
+      c(
+        "Use of mustache patterns {{{{ or }}}} are not allowed in custom components",
```

Comment:
The error message incorrectly displays four braces (`{{{{` and `}}}}`) instead of two. The escaped braces in the string literal will render as `{{{{` and `}}}}` to the user, which is confusing. Change the message to display two braces by using double escaping or by adjusting the glue syntax.
```suggestion
        "Use of mustache patterns {{ or }} are not allowed in custom components",
```